components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1528,6 +1532,7 @@ where
             self.ctx.raft_metrics.propose.read_index += 1;
             let read_proposal = ReadIndexRequest::noop(id, current_time);
             self.fsm.peer.push_pending_read(read_proposal, true);
+            self.fsm.has_ready = true;
```

Comment:
What do you mean by `existing implementations`? 

Also, I think the `has_ready` flag is quite a pitfall, can we replace it with some simple checkings like `self.raft_group.has_ready()` instead?

Reply:
What I mean is use the existing propose method. `has_ready` is not the same as `raft_group.has_ready`. For example, we may rely it to drive snapshot state.

Reply:
`has_ready` should be true because a read request was proposed here.

The reason I don't want to propose read index with existing implementations is that the read may be returned as a follower read.

Reply:
Changed to propose through `propose_raft_command`. (

Reply:
Perhaps `fsm.has_ready` should be removed at all. What `fsm.has_ready` may save is the snapshot and `raft_group.has_ready` checking, both of them should be lightweigth.